app/controllers/users_controller.rb

```diff
@@ -1675,7 +1695,16 @@ def register_passkey
   def delete_passkey
     raise Discourse::NotFound unless SiteSetting.enable_passkeys
 
-    current_user.security_keys.find_by(id: params[:id].to_i)&.destroy!
+    security_key = current_user.security_keys.find_by(id: params[:id].to_i)
+
+    if security_key&.factor_type == UserSecurityKey.factor_types[:first_factor] &&
```

Comment:
```rb
security_key&.factor_type == UserSecurityKey.factor_types[:first_factor]
```

It'd be nice if we could roll this into the `UserSecurityKey` class, as predicate methods, e.g. `security_key&.first_factor?`.